common/djangoapps/student/helpers.py

```diff
@@ -198,6 +220,75 @@ def auth_pipeline_urls(auth_entry, redirect_url=None):
     }
 
 
+def _update_email_opt_in(request, org):
+    """Helper function used to hit the profile API if email opt-in is enabled."""
+
+    email_opt_in = request.POST.get('email_opt_in')
+    if email_opt_in is not None:
+        email_opt_in_boolean = email_opt_in == 'true'
+        preferences_api.update_email_opt_in(request.user, org, email_opt_in_boolean)
+
+
+def enroll(user, course_id, request, check_access):
```

Comment:
@Salomari1987 This refactoring is old. It has been done in 2014, when there was mostly no API in the platform. Please take a look at the [enrollment api](http://edx.readthedocs.io/projects/edx-platform-api/en/latest/enrollment/overview.html).

Take a look at the way I've used the [`update_account_settings`](https://github.com/Edraak/edraak-platform/pull/40/files#diff-8f391fa7e6ff5b7e12c477ca0153f52f) api internally (without having to do a seperate HTTP request).